src/modules/acbs/acbs-facility.service.create-facility.test.ts

```diff
@@ -0,0 +1,260 @@
+import { HttpService } from '@nestjs/axios';
+import { TEST_CURRENCIES } from '@ukef-test/support/constants/test-currency.constant';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import { AxiosError } from 'axios';
+import { when } from 'jest-when';
+import { of, throwError } from 'rxjs';
+
+import { AcbsFacilityService } from './acbs-facility.service';
+import { AcbsCreateFacilityRequest } from './dto/acbs-create-facility-request.dto';
+import { AcbsBadRequestException } from './exception/acbs-bad-request.exception';
+import { AcbsUnexpectedException } from './exception/acbs-unexpected.exception';
+
+describe('AcbsFacilityService', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const idToken = valueGenerator.string();
+  const baseUrl = valueGenerator.httpsUrl();
+  const randomPortfolioIdentifier = valueGenerator.string({ length: 2 });
+  const facilityIdentifier = valueGenerator.stringOfNumericCharacters({ length: 10 });
+
+  let httpService: HttpService;
+  let service: AcbsFacilityService;
+
+  let httpServicePost: jest.Mock;
+
+  const newFacility: AcbsCreateFacilityRequest = {
```

Comment:
This many-line object creation will move to a `...Generator` class in a later PR which will make this file easier to read!